catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImplTest.java

```diff
@@ -199,20 +219,82 @@ public void testToString() {
   }
 
   @Test
-  public void testGetQueryMetacards() {
+  public void testGetEmptyQueryMetacards()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    WorkspaceMetacardImpl workspace = mock(WorkspaceMetacardImpl.class);
+    doReturn("workspaceId").when(workspace).getId();
 
-    Metacard metacard = mock(Metacard.class);
-    when(metacard.getMetacardType()).thenReturn(MetacardImpl.BASIC_METACARD);
+    List<Result> queryResults = Collections.emptyList();
+
+    doReturn(queryResponse).when(catalogFramework).query(any(QueryRequest.class));
+
+    when(queryResponse.getResults()).thenReturn(queryResults);
+
+    List<QueryMetacardImpl> queries = workspaceServiceImpl.getQueryMetacards(workspace);
+
+    assertThat(queries, hasSize(0));
+  }
+
+  @Test
+  public void testGetQueryMetacards()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    WorkspaceMetacardImpl workspace = mock(WorkspaceMetacardImpl.class);
+    doReturn("workspaceId").when(workspace).getId();
+
+    List<Result> queryResults = getMockQueryResults();
+
+    doReturn(queryResponse).when(catalogFramework).query(any(QueryRequest.class));
 
-    String xml = "<xml/>";
+    when(queryResponse.getResults()).thenReturn(queryResults);
 
-    when(workspaceTransformer.xmlToMetacard(xml)).thenReturn(metacard);
+    List<QueryMetacardImpl> queries = workspaceServiceImpl.getQueryMetacards(workspace);
 
-    WorkspaceMetacardImpl workspaceMetacard = mock(WorkspaceMetacardImpl.class);
-    when(workspaceMetacard.getQueries()).thenReturn(Collections.singletonList(xml));
+    assertThat(queries, hasSize(2));
+  }
+
+  @Test
+  public void testGetWorkspaceMetacardFromQueryId()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    WorkspaceMetacardImpl workspace = new WorkspaceMetacardImpl("workspaceId");
+
+    doReturn(queryResponse).when(catalogFramework).query(any(QueryRequest.class));
+
+    List<Result> workspaceResults = Collections.singletonList(getMockResult(workspace));
+    when(queryResponse.getResults()).thenReturn(workspaceResults);
+
+    WorkspaceMetacardImpl workspaceMetacard =
+        workspaceServiceImpl.getWorkspaceFromQueryId("queryId");
+
+    assertThat(workspaceMetacard.getId(), is("workspaceId"));
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void testGetInvalidWorkspaceMetacardFromQueryId()
```

Comment:
This test name is confusing - what exactly is invalid? It looks like this test verifies what happens when the query doesn't return any results.

Reply:
This test covers a scenario where you try to get a workspace from a query id but that workspace could not be found. Changed the method to

`testGetNonExistentWorkspaceMetacardFromQueryID`